tests/circuit_graph/test_tape_circuit_graph.py

```diff
@@ -0,0 +1,203 @@
+        assert circuit.operations[0] is new
+
+    def test_observables(self, circuit, obs):
+        """Test that the `observables` property returns the list of observables in the circuit."""
+        assert circuit.observables == obs
+
+    def test_operations(self, circuit, ops):
+        """Test that the `operations` property returns the list of operations in the circuit."""
+        assert circuit.operations == ops
+
+    def test_op_indices(self, circuit):
+        """Test that for the given circuit, this method will fetch the correct operation indices for
+        a given wire"""
+        op_indices_for_wire_0 = [0, 3, 5, 7]
+        op_indices_for_wire_1 = [1, 3, 6, 8]
+        op_indices_for_wire_2 = [2, 4, 5, 8]
+
+        assert circuit.wire_indices(0) == op_indices_for_wire_0
+        assert circuit.wire_indices(1) == op_indices_for_wire_1
+        assert circuit.wire_indices(2) == op_indices_for_wire_2
+
+    @pytest.mark.parametrize("wires", [['a', 'q1', 3]])
+    def test_layers(self, parameterized_circuit, wires):
+        """A test of a simple circuit with 3 layers and 6 parameters"""
+
+        dev = qml.device("default.gaussian", wires=wires)
+        qnode = qml.QNode(parameterized_circuit, dev)
+        qnode(0.1, 0.2, 0.3, 0.4, 0.5, 0.6)
+        circuit = qnode.qtape.graph
+        layers = circuit.parametrized_layers
+        ops = circuit.operations
+
+        assert len(layers) == 3
+        assert layers[0].ops == [ops[x] for x in [0, 1, 2]]
+        assert layers[0].param_inds == [0, 1, 2]
+        assert layers[1].ops == [ops[3]]
+        assert layers[1].param_inds == [3]
+        assert layers[2].ops == [ops[x] for x in [5, 6]]
+        assert layers[2].param_inds == [6, 7]
+
+    @pytest.mark.parametrize("wires", [['a', 'q1', 3]])
+    def test_iterate_layers(self, parameterized_circuit, wires):
+        """A test of the different layers, their successors and ancestors using a simple circuit"""
+
+        dev = qml.device("default.gaussian", wires=wires)
+        qnode = qml.QNode(parameterized_circuit, dev)
+        qnode(0.1, 0.2, 0.3, 0.4, 0.5, 0.6)
+        circuit = qnode.qtape.graph
+        result = list(circuit.iterate_parametrized_layers())
+
+        assert len(result) == 3
+        assert set(result[0][0]) == set([])
+        assert set(result[0][1]) == set(circuit.operations[:3])
+        assert result[0][2] == (0, 1, 2)
+        assert set(result[0][3]) == set(circuit.operations[3:] + circuit.observables)
+
+        assert set(result[1][0]) == set(circuit.operations[:2])
+        assert set(result[1][1]) == set([circuit.operations[3]])
+        assert result[1][2] == (3,)
+        assert set(result[1][3]) == set(circuit.operations[4:6] + circuit.observables[:2])
+
+        assert set(result[2][0]) == set(circuit.operations[:4])
+        assert set(result[2][1]) == set(circuit.operations[5:])
+        assert result[2][2] == (6, 7)
+        assert set(result[2][3]) == set(circuit.observables[1:])

```

Comment:
Oh, the integers simply represent the parameter index of the circuit. It turns out that tape mode and non-tape mode numbered free parameters differently 😆 

Neither is more correct, so I thought... might as well just update the test to use the new numbering, rather than trying to monkeypatch tape mode to force the same numbering as the old system.